core/src/main/java/org/apache/calcite/sql/SqlBinaryOperator.java

```diff
@@ -171,6 +177,22 @@ private RelDataType convertType(SqlValidator validator, SqlCall call, RelDataTyp
     return type;
   }
 
+  @Nullable Charset getCharsetAfterConvert(SqlBasicCall call, @Nullable Charset typeCharset) {
```

Comment:
Are there other functions which should preserve charsets?
For example, if you do string concatenation, or cast from CHAR to VARCHAR?

Reply:
As far as I know, only CONVERT and TRANSLATE function explicitly use charset.
In MySQL, CAST function can be used like `cast(name as CHAR CHARACTER SET utf8)`, but we don't support such form currently.

Reply:
I was asking about something like `CAST(TRANSLATE("name" using BIG5) AS CHAR(5))`.
Is the charset of the result of this expression correct?

Reply:
The `char(5)` has the charset ISO-8859-1 preserved in SqlCollation, so the charset of `CAST(TRANSLATE("name" using BIG5) AS CHAR(5))` should also be ISO-8859-1.
I add a test for this, please take a look.

Reply:
> I was asking about something like `CAST(TRANSLATE("name" using BIG5) AS CHAR(5))`. Is the charset of the result of this expression correct?

The result charset of `TRANSLATE("name" using BIG5)` is `BIG5`, so the cast to `CHAR(5)` with `ISO-8859-1` is not allowed.